type-tests/@ember/component-test/helper.ts

```diff
@@ -153,17 +133,22 @@ const badReturnSig = helper<DemoSig>(([i18nizer], { name, age }) =>
 );
 
 const greet = helper(([name]: [string]) => `Hello, ${name}`);
+expectTypeOf(greet).toEqualTypeOf<
+  FunctionBasedHelper<{
+    Args: {
+      Positional: [string];
+      Named: object;
+    };
+    Return: string;
+  }>
+>();
 
 // @ts-expect-error
 new greet();
 
 // @ts-expect-error
 class Subgreet extends greet {}
 
+// Check that generics are accepted. (We cannot meaningfully test for how they
+// are *handled* because it uses items we do not want to expose in public API.)
 const pair = helper(<T>([item]: [T]): [T, T] => [item, item]);
-expectTypeOf(pair).toEqualTypeOf<
-  abstract new <T>() => FunctionBasedHelperInstance<{
-    Args: { Positional: [T]; Named: EmptyObject };
-    Return: [T, T];
-  }>
->();
```

Comment:
What will TS produce when emitting declarations for a helper like this if `FunctionBasedHelperInstance` isn't exported?

Reply:
Did some poking and confirmed that we do need this type exported. @dfreeman and I spent a bunch of time discussing alternatives, but ultimately landed on "just export this thing and look forward to when the need for it goes away by dropping `helper()` in favor of *just* using plain functions instead"; done in c725cdce7.